perfkitbenchmarker/publisher.py

```diff
@@ -554,7 +554,11 @@ def AddSamples(self, samples, benchmark, benchmark_spec):
       sample['product_name'] = FLAGS.product_name
       sample['official'] = FLAGS.official
       sample['owner'] = FLAGS.owner
-      sample['timestamp'] = time.time()
+      if 'timestamp' in sample['metadata']:
```

Comment:
I think including it would be helpful. For backwards compatibility, it could be optional, defaulting to the value of `time.time()` at serialization-time.